### ISSUE TITLE
doc.select() hangs on this doc

### ISSUE THREAD

**Issue by indigoviolet** [pymupdf/PyMuPDF#3150]
### Description of the bug

doc.select() hangs indefinitely on this document, for some reason.
[icky-shaggy-gap_Redacted.pdf](https://github.com/pymupdf/PyMuPDF/files/14250922/icky-shaggy-gap_Redacted.pdf)

### How to reproduce the bug

```
In [6]: fitz.open("/Users/venky/Downloads/icky-shaggy-gap_Redacted.pdf").select([0,1,2,3])
^C---------------------------------------------------------------------------
KeyboardInterrupt                         Traceback (most recent call last)
<ipython-input-6-46f416acaa94> in ?()
----> 1 fitz.open("/Users/venky/Downloads/icky-shaggy-gap_Redacted.pdf").select([0,1,2,3])

~/dev/plus/python/.venv/lib/python3.11/site-packages/fitz/__init__.py in ?(self, pyliste)
   5414         # (1) get underlying pdf document,
   5415         # (2) transform Python list into integer array
   5416         pdf = _as_pdf_document(self)
   5417         # call retainpages (code copy of fz_clean_file.c)
-> 5418         retainpages(pdf, pyliste)
   5419         if pdf.m_internal.rev_page_map:
   5420             mupdf.ll_pdf_drop_page_tree(pdf.m_internal)
   5421         self._reset_page_refs()

```

### PyMuPDF version

1.23.21

### Operating system

MacOS

### Python version

3.11

**Comment by indigoviolet** [comment]
(If it turns out to be relevant for prioritization etc, note that I'm filing on behalf of an Enterprise licensee, and can provide further details if needed)

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.23.
